### PR TITLE
API Documentation cleanup

### PR DESCRIPTION
**Pull request by zachcran**
**Is this pull request associated with an issue(s)?**
No.

**Description**
Now that API documentation is online, it needs to be cleaned up and made consistent.

**TODOs**
- [x] Ensure that the "Example Usage" and "Error Checking" sections always appear in the same order.
- [x] Move all brief descriptions to the second line of the block comment. This frees the first line (line containing `#[[[`) for special specifiers like `@module`.
- [x] Move code snippets into a `.. code-block::` so they render properly.
- [x] Proofread documentation.
- [x] Figure out how to render the `:var VARIABLE:` and `:vartype VARIABLE:` commands properly.
- [x] Configure CMinx to remove prefixes from function, macro, and member function parameters.
- [x] Manually remove the parameter prefixes defined in the docstrings.

### CODE REVIEW

**Review by ryanmrichard — APPROVED** [review]

File: cmake/cmakepp_lang/algorithm/equal.cmake [hunk 1]
```diff
@@ -7,35 +7,36 @@ include(cmakepp_lang/types/implicitly_convertible)
...
-# :type _e_lhs: str
-# :param _e_rhs: The other value involved in the comparison
-# :type _e_rhs: str
+# :param result: The name to use for variable holding the result.
+# :type result: desc
+# :param lhs: One of the two values involved in the comparison.
+# :type lhs: str
+# :param rhs: The other value involved in the comparison
+# :type rhs: str
 # :returns: ``_e_result`` will be set to ``TRUE`` if ``_e_lhs`` compares equal
```
Comment (AutonomicPerfectionist) [inline cmake/cmakepp_lang/algorithm/equal.cmake:25]: Missed removal of `_e_` prefix in the returns field. Looks like this same thing was missed in all of the return documentation

**Review by AutonomicPerfectionist — COMMENTED** [review]
@zachcran sorry I didn't get to this sooner. I see it was already merged but figured I'd point out what I saw. Mainly, all of the `:returns:` lines in the documentation still have the prefixes attached. Also, there's an undefined reference to `examples-classes-overriding-object-methods` in `docs/source/features/objects.rst` (Github wouldn't let me comment on that for some reason).

Finally, I noticed there aren't any module doccomments. I looked through the docs action you are using and I think it's using an extremely old version of Sphinx so it wouldn't display the module documentation anyway, but figured I would make you aware just in case.

https://cmakepp.github.io/CMinx/documenting/module.html

File: cmake/cmakepp_lang/algorithm/equal.cmake (hunk 1 above)
Comment (zachcran) [reply cmake/cmakepp_lang/algorithm/equal.cmake:25]: No worries, thanks for pointing this out! I just did a quick find and replace for `:param _[a-zA-Z]+_` and `:type _[a-zA-Z]+_` before submitting, so it makes sense that I missed all of those. I'll have to go back in and fix that.

**Comment by zachcran** [comment]
@AutonomicPerfectionist @ryanmrichard Are module doccomments expected to be at the top of each file, or just the files including all of the files in a subdirectory (for example, `cmake/cmakepp_lang/cmakepp_lang.cmake` and `cmake/cmakepp_lang/algorithm/algorithm.cmake`)

**Comment by AutonomicPerfectionist** [comment]
Top of the file you want to document, before any commands like `include_guard()`

So you can do both: put a module doccomment on the "main" files that include the others, saying what that subdirectory is for, and then have more specific doccomments for each of the individual modules

**Pull request merged**